dagger-common/src/main/java/io/odpf/dagger/common/serde/proto/protohandler/RowFactory.java

```diff
@@ -47,11 +48,25 @@ public static Row createRow(DynamicMessage proto, int extraColumns) {
         Row row = new Row(descriptorFields.size() + extraColumns);
         for (FieldDescriptor fieldDescriptor : descriptorFields) {
             ProtoHandler protoHandler = ProtoHandlerFactory.getProtoHandler(fieldDescriptor);
-            row.setField(fieldDescriptor.getIndex(), protoHandler.transformFromKafka(proto.getField(fieldDescriptor)));
+            row.setField(fieldDescriptor.getIndex(), protoHandler.transformFromProto(proto.getField(fieldDescriptor)));
         }
         return row;
     }
 
+    public static Row createRow(Descriptors.Descriptor descriptor, SimpleGroup simpleGroup, int extraColumns) {
+        List<FieldDescriptor> descriptorFields = descriptor.getFields();
+        Row row = new Row(descriptorFields.size() + extraColumns);
+        for (FieldDescriptor fieldDescriptor : descriptorFields) {
+            ProtoHandler protoHandler = ProtoHandlerFactory.getProtoHandler(fieldDescriptor);
+            row.setField(fieldDescriptor.getIndex(), protoHandler.transformFromParquet(simpleGroup));
+        }
+        return row;
+    }
+
+    public static Row createRow(Descriptors.Descriptor descriptor, SimpleGroup simpleGroup) {
```

Comment:
It will be used in MessageProtoHandler and RepeatedMessageProtoHandler as part of #100 
Can remove it now if you want.

Reply:
> Even in those let's use 0 explicitly
Umm, but why though ? Even for Proto above, we are using a similar constructor

Reply:
As discussed, keeping this,  as this will be needed in the next issue #100 when nested simple groups would need to be converted to Row without any extra columns.